src/sqllogictest/src/runner.rs

```diff
@@ -592,11 +619,44 @@ impl Runner {
             availability_zones: Default::default(),
             connector_context: Default::default(),
         };
-        let server = materialized::serve(mz_config).await?;
-        let client = connect(&server).await;
+        // We need to run the server on its own Tokio runtime, which in turn
+        // requires its own thread, so that we can wait for any tasks spawned
+        // by the server to be shutdown at the end of each file. If we were to
```

Comment:
Why wasn't this a problem with the sqlite stash?

Reply:
The specific problem is a future spawned when you drop a `persist` `ReadHandle` that tries to give up its lease on the shard that it has. Since at the start of every file we drop the existing `persist` data in PostgreSQL, this causes the data to get dropped _before_ that `ReadHandle` (Tokio is slow or something, idk), and then `persist` panics because the shard it is looking for is gone. Didn't happen with SQLite because the file stays live if referenced in process even if it's unlinked on disk. 

Anyway I got scared about this problem in general once I tracked it down. Seems real bad to have futures from prior servers floating around when the new server starts up. Who knows what other things Materialize will learn to spawn on drop over time. (It's a real bummer that `AsyncDrop` doesn't exist.)